perf/src/packet.rs

```diff
@@ -60,6 +60,32 @@ impl PacketBatch {
         batch
     }
 
+    pub fn new_unpinned_with_recycler_data_and_dests<T: Serialize>(
```

Comment:
- Moved inside `PacketBatch`
- `to_packet_batch_with_destination` ==> `new_unpinned_with_recycler_data_and_dests` (to match naming convention of other functions)